BraintreeCore/src/test/java/com/braintreepayments/api/BraintreeGraphQLHttpClientUnitTest.java

```diff
@@ -96,4 +96,46 @@ public void post_withPathAndDataAndConfiguration_sendsHttpRequest() throws Excep
         assertEquals("Bearer encoded_auth_fingerprint", headers.get("Authorization"));
         assertEquals("2018-03-06", headers.get("Braintree-Version"));
     }
+
+    @Test
+    public void post_withPathAndDataAndConfigurationAndCallback_withInvalidToken_forwardsExceptionToCallback() {
+        Authorization authorization = new InvalidToken("invalid", "token invalid");
+
+        BraintreeGraphQLHttpClient sut = new BraintreeGraphQLHttpClient(authorization, httpClient);
+        sut.post("sample/path", "data", configuration, httpResponseCallback);
+
+        ArgumentCaptor<BraintreeException> captor = ArgumentCaptor.forClass(BraintreeException.class);
+        verify(httpResponseCallback).failure(captor.capture());
+
+        BraintreeException exception = captor.getValue();
+        assertEquals("token invalid", exception.getMessage());
+    }
+
+    @Test
+    public void post_withDataAndConfigurationAndCallback_withInvalidToken_forwardsExceptionToCallback() {
+        Authorization authorization = new InvalidToken("invalid", "token invalid");
+
+        BraintreeGraphQLHttpClient sut = new BraintreeGraphQLHttpClient(authorization, httpClient);
+        sut.post("sample/path",  configuration, httpResponseCallback);
+
+        ArgumentCaptor<BraintreeException> captor = ArgumentCaptor.forClass(BraintreeException.class);
+        verify(httpResponseCallback).failure(captor.capture());
+
+        BraintreeException exception = captor.getValue();
+        assertEquals("token invalid", exception.getMessage());
+    }
+
+    @Test
+    public void post_withPathAndDataAndConfiguration_withInvalidToken_throwsBraintreeException() throws Exception {
```

Comment:
```suggestion
    public void post_withPathAndDataAndConfiguration_withInvalidToken_throwsBraintreeException() {
```
It looks like that might not be needed since we're catching the exception in the test?

Reply:
Good catch - Removed the unnecessary throws in this test file 👍